src/lightning/data/streaming/dataset_optimizer.py

```diff
@@ -798,3 +837,38 @@ def _upload_index(self, cache_dir: str, num_nodes: int, node_rank: Optional[int]
             merge_cache = Cache(cache_dir, chunk_bytes=1)
             merge_cache._merge_no_wait()
             self._upload_index(cache_dir, 1, None)
+
+    def _cleanup_cache(self) -> None:
+        cache_chunks_dir = os.path.join(_get_cache_folder(), self.name)
+
+        # Cleanup the cache folder to avoid corrupted files from previous run to be there.
+        if os.path.exists(cache_chunks_dir):
+            rmtree(cache_chunks_dir)
+
+        cache_data_dir = os.path.join(_get_cache_folder(), "data", self.name)
```

Comment:
You should unify that path somewhere, in a function/method. Otherwise you will change it in one place and forget to change here, and then `os.path.exists(cache_data_dir)` will always be False. Or you write a unit test for it, or both ;)

Reply:
Sure thing ;)